test/functional/services/common/browser.ts

```diff
@@ -479,11 +479,29 @@ export async function BrowserProvider({ getService }: FtrProviderContext) {
       );
     }
 
-    public async executeAsync<R>(
-      fn: string | ((...args: any[]) => Promise<R>),
+    public async executeAsync<T = unknown>(
+      fn: (cb: (value?: T) => void) => void | Promise<void>
+    ): Promise<T>;
```

Comment:
Isn't the signature strictly ` fn: (cb: (value?: T) => void) => void`? Do we need the `| Promise<void>` ?